x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/jwt/JwtRealmSettings.java

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.core.security.authc.jwt;
+
+import org.apache.http.HttpHost;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.util.set.Sets;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.xpack.core.security.authc.RealmSettings;
+import org.elasticsearch.xpack.core.security.authc.support.ClaimSetting;
+import org.elasticsearch.xpack.core.security.authc.support.DelegatedAuthorizationSettings;
+import org.elasticsearch.xpack.core.ssl.SSLConfigurationSettings;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+
+public class JwtRealmSettings {
+
+    private JwtRealmSettings() {}
+
+    public static final String TYPE = "jwt";
+    private static final String CLIENT_AUTHENTICATION_PREFIX = RealmSettings.PREFIX + TYPE + ".client_authentication.";
+
+    public static final List<String> SUPPORTED_SIGNATURE_ALGORITHMS = List.of(
+        "HS256",
+        "HS384",
+        "HS512",
+        "RS256",
+        "RS384",
+        "RS512",
+        "ES256",
+        "ES384",
+        "ES512",
+        "PS256",
+        "PS384",
+        "PS512"
+    );
+
+    public static final List<String> SUPPORTED_CLIENT_AUTHENTICATION_TYPE = List.of("sharedsecret", "ssl", "none");
```

Comment:
does it make sense to name this client authentication type `mtls` or `mutual_tls`  instead ? But also introduce this in the separate PR where we will be introducing client tls authentication support ?